### ISSUE TITLE
Change libbpfgo map methods to use unsafe.Pointer

### ISSUE THREAD

**Issue by grantseltzer** [aquasecurity/tracee#814]
In https://github.com/aquasecurity/libbpfgo/pull/32 the `Update`, `DeleteKey`, and `GetValue` methods of BPFMap were changed to use unsafe.Pointer's instead of `interface{}`s.

**Comment by grantseltzer** [comment]
@itaysk I want to make sure this is in the next release of tracee. This is a blocker from upgrading tracee to new versions of libbpfgo

**Comment by rafaeldtinoco** [comment]
![image](https://user-images.githubusercontent.com/7395852/127369714-fb11909f-17fd-4185-9c0b-da405f988545.png)

I had to revert those changes locally to keep local sources of tracee and libbpfgo (so I can develop both together). I started making the changes but saw they would require some changes in setUintFilter (for UID, PID, ..., and anything using consts passed as arguments to setXXXFilter) then dropped it, and then Grant told me about this issue.

**Comment by rafaeldtinoco** [comment]
Plan here is:

- pull request #864 finished (Fixes: #862), add more commits:
- tracee needs to use latest libbpfgo (Fixes: #894)
- change tracee to use new libbpfgo map methods (Fixes: #814)

I'm working on it and will pull #864 altogether soonish.

**Comment by rafaeldtinoco** [comment]
For this change we will have to do some modifications in the way we rely in 'costs' for configuration sets into maps. 

Example:

```go
const (
	filterIn  uint8 = 1
	filterOut uint8 = 2
)
```

```go
	configFilteru32 := uint32(configFilter)
	configFilterIn := uint32(filterIn)
	configFilterOut := uint32(filterOut)

	bpfConfigMap, err := t.bpfModule.GetMap("config_map") // u32, u32
	if err != nil {
		return err
	}
	if len(filter.Equal) > 0 && len(filter.NotEqual) == 0 {
		bpfConfigMap.Update(unsafe.Pointer(&configFilteru32), unsafe.Pointer(&configFilterIn))
	} else {
		bpfConfigMap.Update(unsafe.Pointer(&configFilteru32), unsafe.Pointer(&configFilterOut))
	}
```

Instead of creating local variables every time we need to call map.Update() function.. we can, instead of relying in consts, rely in initialized object variables with the same value. This way they will always have stack addresses to unsafe.Pointer() point at.

Because of the what is explained at https://github.com/aquasecurity/tracee/issues/814#issuecomment-899001156, I might do the local variables initially and open a new issue to refactor it in more appropriate time (re-thinking the consts approach for bpfmap calls).
